### ISSUE TITLE
Fail to detect usage of unsupported when used as method references

### ISSUE THREAD

**Issue by csoroiu** [mojohaus/animal-sniffer#22]
I am using **animal-sniffer-maven-plugin** version 1.15 with **java17** signature.
I have the following method:
```java
public <T> void foo() {
    com.acme.Function<Optional<T>, T> v = Optional::get;
}
```
`com.acme.Function` does not extend `java.util.function.Function`.
the `Optional` from `Optional::get` method reference does not get detected.

But, when converting the method reference to a lambda, detection works fine:
```java
public <T> void bar() {
    com.acme.Function<Optional<T>, T> v = o -> o.get();
}
```

---
As far as I can tell, the `MethodVisitor` returned by `SignatureChecker.visitMethod` method should override the `MethodVisitor.visitInvokeDynamicInsn` method.

**Comment by csoroiu** [comment]
this issue is fixed by #40 PR.
